### DISCUSSION TITLE
self hosting with node js and MySQL

### DISCUSSION THREAD

**Discussion by ethan1141** [requarks/wiki#6615] [Help / Questions]
Hi,

I can't create the db with node server command but it fails due to the default db engine not being innodb, it is myisam. The support team can't change the default as it will set server wide. 

I created it on my localhost and uploaded the db that works fine but when I run node server, it finds and finishes doing its thing.

It gets stuck on rebuild-tree. Can anyone help?

![image](https://github.com/requarks/wiki/assets/53418441/6744de17-48e3-4c80-8103-dfa76ef92c5c)

**Comment by NGPixel** [comment]
You can set a different MySQL engine per table / database... and even change it on existing tables.

Also, I doubt the engine used is the issue here. Which version of MySQL are you using?

**Reply by ethan1141** [reply]
Thanks for replying 

Server version: 10.6.14-MariaDB - MariaDB Server

Specified key was too long; max key length is 1000 bytes - this first error was on the analytics table with node server

Node v16.20.1 as per the docs

**Reply by ethan1141** [reply]
That works on my uploaded table to Innodb but when I run node server it has rebuilding tree
